core/src/main/java/org/mule/api/processor/CompletionHandler.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.api.processor;
+
+import org.mule.api.MessagingException;
+
+/**
+ * Interfaces used to perform callback in the case of completion of processing or failure.
+ */
+public interface CompletionHandler<T, E>
```

Comment:
"Interfaces used to perform.. " -> "Performs a callback"
If this is used only on non blocking, then that should be added into the javadoc